### PR TITLE
is_isotropic and is_isotropic_with_vector for rational quadratic spaces 

### PR DESCRIPTION
**Pull request by simonbrandhorst**
Use indefinite lll to solve a quadratic form.

### CODE REVIEW

File: src/QuadForm/Herm/Spaces.jl [hunk 1]
```diff
@@ -53,6 +53,13 @@ function hermitian_space(E::NumField, gram::MatElem; cached::Bool = true)
   return HermSpace(E, K, gramc, involutionV, cached)
 end
 
+function rescale(V::HermSpace, r, cached::Bool=true)
+  E = base_ring(V)
+  K = fixed_field(V)
+  r = K(r)
+  return HermSpace(E, K, r*gram_matrix(q), involution(V), cached=cached)
```
Comment (thofma) [inline src/QuadForm/Herm/Spaces.jl:60]: Hm, there is no `q` (and it is not covered by the tests).

File: src/QuadForm/Quad/Spaces.jl [hunk 2]
```diff
@@ -816,6 +824,9 @@ end
 # p is the prime idela
 function _can_locally_embed(n::Int, da, ha::Int, m::Int, db, hb::Int, p)
   de = m - n
+  if de < 0
+    return false
```
Comment (thofma) [inline src/QuadForm/Quad/Spaces.jl:828]: Did you have an example for this?

File: src/QuadForm/Quad/Spaces.jl [hunk 3]
```diff
@@ -1162,9 +1173,89 @@ end
...
 _to_gf2(x) = x == 1 ? 0 : 1
-#ToGF2:= func< a,b,p | HilbertSymbol(a,b,p) eq 1 select 0 else 1 >;
-#SignGF2:= func< x, p | Evaluate(x, p) lt 0 select 1 else 0 >;
-#MyFact:= func< R, d | Type(R) eq RngInt select FactorizationOfQuotient(Rationals() ! d) else Factorization(R*d) >;
+
+function is_isotropic_with_vector(q::QuadSpace{FlintRationalField, fmpq_mat})
+  ok, S = _isotropic_subspace(q)
+  if !ok
+    z = zeros(base_ring(q), 1, dim(q))
+    return false, z
```
Comment (thofma) [inline src/QuadForm/Quad/Spaces.jl:1181]: Would be good to add a non-isotropic test.

**Comment by thofma** [comment]
Do you know how the runtime imroved?

File: src/QuadForm/Quad/Spaces.jl (hunk 2 above)
Comment (simonbrandhorst) [reply src/QuadForm/Quad/Spaces.jl:828]: The logic should be clear if dim(M)=m > n=dim(N) then M cannot embedd in N

Comment (thofma) [reply src/QuadForm/Quad/Spaces.jl:828]: Yes, but it is not covered by the tests.

**Comment by simonbrandhorst** [comment]
Before
```
F = QQ[2 0 0 0 0 0; 0 1 0 0 0 0; 0 0 70//9 0 0 0; 0 0 0 -311//105 64//21 -286//105; 0 0 0 64//21 -67//21 65//21; 0 0 0 -286//105 65//21 -446//105];

julia> @btime fl, v = Hecke._isisotropic_with_vector(F)
  59.440 ms (654117 allocations: 34.14 MiB)
(true, fmpq[8326838191303405660211848320, 0, 1619442456198070214901, 6842437510954948584163537055, 19317415442821417874855, 0])
```

After
```
julia> F = QQ[2 0 0 0 0 0; 0 1 0 0 0 0; 0 0 70//9 0 0 0; 0 0 0 -311//105 64//21 -286//105; 0 0 0 64//21 -67//21 65//21; 0 0 0 -286//105 65//21 -446//105];

julia> q = quadratic_space(QQ, F);

julia> @btime fl, v = Hecke.is_isotropic_with_vector(q)

  2.442 ms (27050 allocations: 1.44 MiB)
(true, fmpq[0, -1//42, 0, 11//42, 19//63, 1//18])
```
So it is a factor of 20 + the result has smaller coefficients.

File: src/QuadForm/Herm/Spaces.jl [hunk 4]
```diff
@@ -53,6 +53,13 @@ function hermitian_space(E::NumField, gram::MatElem; cached::Bool = true)
   return HermSpace(E, K, gramc, involutionV, cached)
 end
 
+function rescale(V::HermSpace, r, cached::Bool=true)
+  E = base_ring(V)
+  K = fixed_field(V)
+  r = K(r)
+  return HermSpace(E, K, r*gram_matrix(V), involution(V), cached=cached)
```
Comment (thofma) [inline src/QuadForm/Herm/Spaces.jl:60]:
```suggestion
  return HermSpace(E, K, r*gram_matrix(V), involution(V), cached)
```

**Pull request merged**